text/0000-add-scratch-stack.md

```diff
@@ -0,0 +1,161 @@
+in-turn, make Paketo less attractive for developers of these applications.
+
+Creating a scratch stack will decouple these use cases, allowing us to tailor
+the tiny stack to applications which want more packages, while also providing a
+better experience for statically-linked binaries.
+
+## Detailed Explanation
+
+### Repository/Image Location
+
+We will create a new Github repository at
+`paketo-buildpacks/jammy-scratch-stack` and a new pair of images on Dockerhub
+at `paketobuildpacks/build-jammy-scratch` and
+`paketobuildpacks/run-jammy-scratch`.
+
+We will adopt the tag system used for Jammy - specifically semver (i.e.
+'X.Y.Z') and 'latest' tags.
+
+We will not create a builder repository or image, though we reserve the option
+for the Builder team to do so in the future if that is requested and approved.
+
+### Stack ID
+
+We propose creating a new stack ID: `io.buildpacks.stacks.jammy.scratch`.
+This may require approval from the upstream Cloud-Native Buildpacks project.
+
+### Image Contents
+
+The scratch stack run image will contain the `tzdata` and `ca-certificates`
+packages, the standard CNB/Paketo metadata (including users and /etc/passwd
+entries), and nothing else. These two packages are often provided in scratch
+images as they are small and have minimal security risk.
+
+The build image will contain the same packages as the tiny stack (probably with
+the addition of the packages proposed in [the tiny packages
+RFCS](https://github.com/paketo-buildpacks/rfcs/pull/231/files)).
+
+In practice, the scratch stack will look like a stripped-down version of the
+existing tiny stack, with most packages removed and minor tweaks to the
+metadata to support the change in name, support URL, etc.
+
+### Automation
+
+We will leverage the existing stack automation that all the Paketo stacks use.
+This includes checking for USNs on a regular cadence and automatically pushing
+new releases to Dockerhub as they are created.
+
+Although there are very few CVEs for the `tzdata` and `ca-certificates`
+packages, and hence the run image will rarely be updated, we want to ensure the
+build image is also kept up to date with CVE fixes for all of its packages.
+
+Also, on a practical note, leveraging the existing automation that is used in
+all other stacks is simpler for the stacks maintainers than creating a new
+system just for this stack.
+
+### Golang Buildpack
+
+We will add support for this new stack in the golang buildpack. This will
+enable the following:
+
+1. Golang application developers can utilize this new stack out-of-the-box,
+   without having to modify the Golang buildpack to support this new stack.
+1. The scratch stack can use the Golang buildpack for its integration tests.
+   This will enable meaningful end-to-end integration tests for this stack. We
+   want to be able to build and run a test application during the stack
+   integration test, and golang is the best choice for this.
+
+## Rationale and Alternatives
```

Comment:
I don't think that it would be a good idea to make different images under the same stack ID because I think it would be confusing to say: "this buildpack works with the XYZ stack except if you provide XYZ stack with ABC run image swapped out".

Maybe my mental model of stacks is wrong, but I've found it helpful to think of stacks as a set of interfaces - i.e. a list of ABI-compatible libraries on top of a specific distribution. I know that full and base have the same stack IDs and honestly I think that's confusing too.

I think it is more intuitive to be able something like the following:
* go and rust work on `static`, `tiny`, `base`, and `full`
* java (and hopefully soon nodejs too) work on `tiny`, `base`, and `full`
* python works on `base`, and `full`
* php works on only on `full`

I think the best way to express that compatibility relationship between buildpacks and stacks is programmatically, which means in the `buildpack.toml`.

Also, practically, the tooling we have today won't make it particularly easier to achieve what you described. `jam create-stack` will still create a stack with a pair of images, so we'd have to discard the build image at the builder level.

I think this comes down to what one considers to be the interface of a given stack identifier. Perhaps you have a different set of expectations around stacks?